cf_units/__init__.py

```diff
@@ -781,7 +781,7 @@ def as_unit(unit):
         if use_cache:
             result = _CACHE.get(unit)
         if result is None:
-            result = Unit(unit)
```

Comment:
Perhaps a comment stating that typically unit is a string, but it could be another type of Unit (e.g. iris.unit.Unit).